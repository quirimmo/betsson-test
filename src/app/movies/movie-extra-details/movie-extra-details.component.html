<section class="row movie-overview-wrapper" [@bounceInUpOnEnter]>
	<div class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 mb-3">
		{{ movieDetails.overview }}
	</div>
	<div
		class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-release-date mb-3 text-center"
	>
		Release date: {{ movieDetails.releaseDate | date: 'fullDate' }}
	</div>
	<div
		class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3 text-center"
	>
		Homepage:
		<a class="hvr-underline-from-center" href="{{ movieDetails.homepage }}">{{
			movieDetails.homepage
		}}</a>
	</div>
	<div
		class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3"
	>
		<div class="row">
			<div class="col-sm-6 text-center movie-budget">
				Budget: {{ movieDetails.budget | currency }}
			</div>
			<div class="col-sm-6 text-center movie-revenue">
				Revenue: {{ movieDetails.revenue | currency }}
			</div>
		</div>
	</div>
	<div
		class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3"
	>
		<div class="row mb-1" *ngIf="movieDetails.videos.length === 0">
			<div class="col-12 text-center">THERE ARE NOT AVAILABLE VIDEOS</div>
		</div>
		<div class="row mb-1" *ngIf="movieDetails.videos.length > 0">
			<div class="col-12 text-center">AVAILABLE VIDEOS</div>
		</div>
		<div class="row mb-2" *ngFor="let video of movieDetails.videos">
			<div class="col-12 text-center">{{ video.name }}</div>
			<div class="col-12 text-center">
				<a
					class="hvr-underline-from-center"
					href="https://www.youtube.com/watch?v={{ video.key }}"
					>https://www.youtube.com/watch?v={{ video.key }}</a
				>
			</div>
		</div>
	</div>
</section>
