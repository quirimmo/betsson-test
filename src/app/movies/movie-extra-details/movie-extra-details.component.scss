@import './../../../colors';
.movie-overview-wrapper {
  color: $grey;
  font-size: 13pt;
  margin-top: 70px;
  text-align: justify;
}

.movie-release-date {
  color: $yellow;
  font-size: 13pt;
  letter-spacing: 1;
}

.movie-budget {
  color: $red;
}

.movie-revenue {
  color: $green;
}

a {
  color: $blue;
}

a:hover {
  text-decoration: none;
}

.hvr-underline-from-center:before {
  background: $blue;
  height: 2px;
}
