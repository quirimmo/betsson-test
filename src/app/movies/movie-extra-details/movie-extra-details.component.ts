import { Component, Input } from '@angular/core';
import { MovieDetails } from '../movie-details.model';
import { bounceInUpOnEnterAnimation } from 'angular-animations';

@Component({
	selector: 'movie-extra-details',
	inputs: ['movie-details'],
	styleUrls: ['./movie-extra-details.component.scss'],
	templateUrl: './movie-extra-details.component.html',
	animations: [bounceInUpOnEnterAnimation({ delay: 600, duration: 400 })]
})
export class MovieExtraDetailsComponent {
	@Input('movie-details')
	movieDetails: MovieDetails;

	constructor() {}
}
