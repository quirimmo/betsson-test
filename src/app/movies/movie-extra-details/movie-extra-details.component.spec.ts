import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { MovieExtraDetailsComponent } from './movie-extra-details.component';

const declarations = [MovieExtraDetailsComponent];
const providers = [];
const imports = [];

describe('MovieExtraDetails Component', () => {
	let componentFixture: ComponentFixture<any>, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		}).compileComponents();
		componentFixture = TestBed.createComponent(MovieExtraDetailsComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
	}));

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});
});
