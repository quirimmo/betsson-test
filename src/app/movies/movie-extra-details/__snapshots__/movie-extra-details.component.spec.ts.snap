// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieExtraDetails Component should match the snapshot 1`] = `
<movie-extra-details>
  <section
    class="row movie-overview-wrapper"
  >
    <div
      class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 mb-3"
    >
       
    </div>
    <div
      class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-release-date mb-3 text-center"
    >
       Release date: 
    </div>
    <div
      class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3 text-center"
    >
       Homepage: 
      <a
        class="hvr-underline-from-center"
      >
        
      </a>
    </div>
    <div
      class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-6 text-center movie-budget"
        >
           Budget: 
        </div>
        <div
          class="col-sm-6 text-center movie-revenue"
        >
           Revenue: 
        </div>
      </div>
    </div>
    <div
      class="col-xl-6 offset-xl-3 col-md-8 offset-md-2 col-12 movie-homepage mb-3"
    >
      
      
      
    </div>
  </section>
</movie-extra-details>
`;
